src/components/NewsCard/NewsCard.js

```diff
@@ -20,50 +40,76 @@ function NewsCard(props) {
       setactiveTag('');
     }
   }, [trash, bookmark]);
+
   const handleMouseEnter = () => {
-    setIsShown(true);
-    setBookmark('card__bookmark-bold');
-    setTrash('card__trash-bold');
+    if (bookmark !== 'card__bookmark-marked') {
+      if(icon.name === 'Trash' || !user.isLoggedIn) setIsShown(true);
+      setBookmark('card__bookmark-bold');
+      setTrash('card__trash-bold');
+    }
   };
   const handleMouseLeave = () => {
-    setIsShown(false);
-    setBookmark('card__bookmark');
-    setTrash('card__trash');
+    if (bookmark !== 'card__bookmark-marked') {
+      setIsShown(false);
+      setBookmark('card__bookmark');
+      setTrash('card__trash');
+    }
+  };
+
+  const HandleCardIconClick = (id) => {
+    if (user.isLoggedIn && location === '/') {
+      mainApi
+        .addArticle({
+          keyword: props.currentKeyword,
+          title,
+          text: description,
+          source: source.name,
+          image: urlToImage,
+          date: publishedAt,
+          link: url,
+        })
+        .then((res) => {
+          setBookmark('card__bookmark-marked');
+          setDisabled(true);
+        });
+    } else {
+      props.HandleCardIconClick(id);
+    }
   };
 
   return (
-    <li className='card'>
+    <li className='card' id={_id}>
       <article className='card__item'>
         <button
           src={icon.name}
+          onClick={() => HandleCardIconClick(_id)}
```

Comment:
`          onClick={() => HandleCardIconClick(_id)}
`
**Might be better**

the criteria say: **Arrow functions are not passed as handler functions.**